File: QuickApp/ClientApp/src/app/models/states.ts
export class states {

    constructor(Country_Name?:string,States_Country_ID?:number,districtId?:number,States_ID?: number, States_Name?: string) {

      
        this.States_ID = States_ID;
        this.States_Name = States_Name;
        this.Country_Name=Country_Name;
        this.States_Country_ID=States_Country_ID;
        this.districtId=districtId;
    }

    
    public States_ID: number;
    public States_Name: string;
    public Country_Name: string;
    public States_Country_ID: number;
    public districtId:number=0;
    

}

File: DAL/Models/Contact Information/States.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DAL.Models.Contact_Information
{
    public class States :AuditableEntity
    {
        public int States_ID { get; set; }
        public string States_Name { get; set; }
        //foreign key from country
        public int States_Country_ID { get; set; }

        public int? districtId { get; set; }

        public ICollection<Address> address { get; set; }

        public ICollection<City> city { get; set; }

        public Country country { get; set; }

        public ICollection<District> district { get; set; }

    }
}
